sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -74,10 +74,22 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   }
 
   /**
-   * @return All metrics containing metrics of this SparkPlan.
+   * @return Executor side metrics of this SparkPlan, which needs to be serialized and sent
+   *         to executors.
    */
   def metrics: Map[String, SQLMetric] = Map.empty
 
+  /**
+   * @return Driver side metrics of this SparkPlan, which only create, update and display
+   *         on driver side, no need to be serialized and sent to executors.
```

Comment:
also mention that, when override this method, people should use `@transient override lazy val`

Reply:
Thanks for remind, done in this commit.